### DISCUSSION TITLE
Coupon codes that can only be used one a customer's first order

### DISCUSSION THREAD

**Discussion by eatigc** [swellstores/community#212] [Help] [answered]
Is there a way to create a coupon that can only be used on a customer's first order?

I'm trying to implement a customer referral program.

Currently, whenever a user places their first order I have a web hook that creates unique referral code as a coupon for each customer so that they can pass it to new customers. 

I want the new customer to be able to use the referral code and get a discount on their first order, whether they create from an anonymous cart or whether they apply it to a cart while logged in - as long as they have not yet placed any orders.

The reason I mention an anonymous cart is I'm currently limited the referral codes to a group called "first_time_users"
and I have a webhook on account creation that places each new account automatically  into this group.

I've written code on my custom storefront that automatically applies the coupon to the cart when they access the website using a unique link, but if the user is not logged in, the coupon can't be applied because the anonymous user is not in the "first_time_users" group and currently, I don't have any other way to limit the use of the coupon to only work for customers who don't have an order yet

If there was an inverse rule, like, this coupon can only be used if a customer is NOT in a certain group, maybe that would work better. But that doesn't seem to exist right now.

Any help would be much appreciated!

**Comment by sags95** [comment]
When creating coupons, there is the option to generate multiple codes for this specific coupon. In this case you can limit the usage to one use per unique code, so it would work regardless if it's linked to a customer or not. Perhaps in this case they would receive one of these codes as part of the referral?

<img width="1202" alt="Screenshot 2023-04-12 at 2 53 36 PM" src="https://user-images.githubusercontent.com/9522171/231557841-ad97ac82-2fa0-4c2d-80e5-07ac7ddbc229.png">

**Reply by eatigc** [reply]
the problem is that the user would be able to use it for orders beyond their first order, and its supposed to be an introductory offer for new users

**Reply by sags95** [reply]
In this case you would not be able to reuse the same code, as an error would be thrown when trying to apply it to another cart.

**Reply by eatigc** [reply]
is there any way to have a coupon that only applies to a user's first order?

**Reply by sags95** [reply]
@eatigc you can also apply the "one use per customer" setting.

**Reply by logeshswell** [reply]
No, I guess `one use per customer` setting still doesn't solve this usecase.

What if the customer already placed the first order without the coupon?
Then the valid coupon should throw the error (saying that this coupon is valid for first-time orders) which is what @eatigc is expecting if I am not wrong.

**Comment by eatigc** [comment]
@sags95 is there a way to make a coupon not usable by a certain group?

**Reply by logeshswell** [reply, marked as answer]
No, there isn't a direct solution, but you can implement your coupon acceptance criteria using real-time order webhook following the below guide:
https://developers.swell.is/guides/core-concepts/webhook-basics#real-time-order-webhooks
